src/datadog/common/hash.cpp

```diff
@@ -0,0 +1,317 @@
+#include "hash.h"
+
+#include <memory.h>
+
+#if _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4267)
+#endif
+
+constexpr bool ALLOW_UNALIGNED_READS = true;
+
+namespace datadog::common {
+namespace details {
+
+void SpookyHash::Short(const void *message, size_t length, uint64 *hash1,
```

Comment:
Any reason to use spooky versus other hash implementations like [xxh3](https://github.com/Cyan4973/xxHash?tab=readme-ov-file) ? 
Feel free to discard the question if this is not so relevant.

Reply:
Hey @r1viollet , thanks for chiming in. I must admit I wasn't aware of xxhash. I favored SpookyHash mainly because the performance for long strings is slightly better than CityHash, and the append interface makes it less error-prone for combining hashes. I'll take a look at xxhash. Thanks!